ui/ducks/metamask/metamask.js

```diff
@@ -198,6 +210,49 @@ export default function reduceMetamask(state = {}, action) {
   }
 }
 
+const toHexWei = (value, expectHexWei) => {
+  return addHexPrefix(expectHexWei ? value : decGWEIToHexWEI(value));
+};
+
+// Action Creators
+export function updateTransactionGasFees({
+  gasPrice,
+  gasLimit,
+  maxPriorityFeePerGas,
+  maxFeePerGas,
+  isModal,
+  transaction,
+  expectHexWei = false,
+}) {
+  return async (dispatch) => {
+    const txParamsCopy = { ...transaction.txParams, gas: gasLimit };
+    if (gasPrice) {
+      dispatch(
+        setCustomGasPrice(toHexWei(txParamsCopy.gasPrice, expectHexWei)),
+      );
+      txParamsCopy.gasPrice = toHexWei(gasPrice, expectHexWei);
+    } else if (maxFeePerGas && maxPriorityFeePerGas) {
+      txParamsCopy.maxFeePerGas = toHexWei(maxFeePerGas, expectHexWei);
+      txParamsCopy.maxPriorityFeePerGas = addHexPrefix(
+        decGWEIToHexWEI(maxPriorityFeePerGas),
+      );
+    }
+    const updatedTx = {
+      ...transaction,
+      txParams: txParamsCopy,
+    };
+    dispatch(setCustomGasLimit(addHexPrefix(gasLimit.toString(16))));
```

Comment:
this should be moved down into the isModal block, along with the setCustomGasPrice call which are only needed if in the isModal flow (old code)